### PR TITLE
remove NaturalTransformation.apply

### PR DESCRIPTION
**Pull request by xuwei-k**
I think NaturalTransformation is not typeclass.
`NaturalTransformation.apply` should not exists

### CODE REVIEW

File: core/src/main/scala/cats/arrow/NaturalTransformation.scala [hunk 1]
```diff
@@ -6,5 +6,4 @@ trait NaturalTransformation[F[_], G[_]] {
 }
 
 object NaturalTransformation {
-  def apply[F[_], G[_]](implicit ev: NaturalTransformation[F, G]): NaturalTransformation[F, G] = ev
```
Comment (julien-truffaut) [inline core/src/main/scala/cats/arrow/NaturalTransformation.scala]: should you remove the companion object then?

**Comment by ceedubs** [comment]
I'm inclined to agree with the notion that NaturalTransformation shouldn't be thought of as a type class. Is there anyone that strongly feels differently?

**Comment by non** [comment]
Right this makes sense. :+1: 

I thought of this as a way to get an implicit `A ~> B` but I agree that it's probably not needed.

**Comment by ceedubs** [comment]
:+1: for merging (and we can remove the companion object entirely as well probably)

**Comment by xuwei-k** [comment]
removed companion object

**Pull request merged**